fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -1749,11 +1752,24 @@ private void sendStmtPrepareOK(PrepareStmt prepareStmt) throws IOException {
                 serializer.writeField(colNames.get(i), parameters.get(i).getType());
                 context.getMysqlChannel().sendOnePacket(serializer.toByteBuffer());
             }
-            context.getState().setEof();
-        } else {
-            context.getMysqlChannel().flush();
-            context.getState().setStateType(MysqlStateType.NOOP);
+            MysqlEofPacket eofPacket = new MysqlEofPacket(context.getState());
+            eofPacket.writeTo(serializer);
+            context.getMysqlChannel().sendOnePacket(serializer.toByteBuffer());
         }
+
+        if (numColumns > 0) {
```

Comment:
why not make it `else if` clause to make people know that only 1 scenario will happen.